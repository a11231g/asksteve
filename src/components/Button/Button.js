import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { TouchableOpacity, View, ActivityIndicator, Text } from 'react-native';
import styles from './ButtonStyle';

export default class Button extends PureComponent {
    static propTypes = {
        type: PropTypes.oneOf(['primary', 'secondary', 'success']),
        children: PropTypes.any.isRequired,
        onPress: PropTypes.func,
        loading: PropTypes.bool,
        disabled: PropTypes.bool,
        style: PropTypes.objectOf(PropTypes.any),
        size: PropTypes.oneOf(['small', 'medium']),
    };

    static defaultProps = {
        loading: false,
        disabled: false,
        type: 'primary',
        style: {},
        size: 'small',
        children: ''
    };

    handlePress = () => {
        const { loading, onPress, disabled } = this.props;
        if (!loading && !disabled && onPress) {
            onPress();
        }
    };
    renderButton = () => {
        const { type, disabled, loading, children, style, size } = this.props;
        if (disabled) {
            return (<View
                style={[styles.default, styles[type], style, { opacity: 0.2 }]}
            >
                <View style={[styles.defaultContent, styles[`${type}Content`]]}>
                    {
                        loading ? <ActivityIndicator color="#fff" /> :
                            <View style={styles.contentWrapper}>
                                <Text style={[styles[`${type}ContentText`], styles[`${size}ContentText`]]}>{children}</Text>
                            </View>
                    }
                </View>
            </View>);
        }
        return (<TouchableOpacity
            style={[styles.default, styles[type], style]}
            onPress={this.handlePress}
        >
            <View style={[styles.defaultContent, styles[`${type}Content`]]}>
                {
                    loading ? <ActivityIndicator color="#fff" /> :
                        <View style={[styles.defaultContent, styles[`${type}Content`]]}>
                            <Text style={[styles[`${type}ContentText`], styles[`${size}ContentText`]]}>{children}</Text>
                        </View>
                }
            </View>
        </TouchableOpacity>);
    };

    render() {
        return (
            this.renderButton()
        );
    }
}
