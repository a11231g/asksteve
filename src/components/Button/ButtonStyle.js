import { StyleSheet } from 'react-native';


export default StyleSheet.create({
    default: {
        height: 40,
        alignItems: 'center',
        justifyContent: 'center',
        padding: 8,
        borderRadius: 3,
    },
    primary: {
        backgroundColor: '#1e87f0',
        borderRadius: 3,
    },
    small: {
        height: 40,
    },
    medium: {
        height: 50,
    },
    secondary: {
        backgroundColor: '#fff',
        borderWidth: 1,
        borderColor: '#1e87f0',
        borderRadius: 3
    },
    success: {
        backgroundColor: '#26c36c'
    },
    defaultContent: {
        textAlign: 'center'
    },
    primaryContentText: {
        color: '#eee',
        fontSize: 16
    },
    secondaryContentText: {
        color: '#1e87f0',
        fontSize: 16

    },
    successContentText: {
        color: '#ffffff',
        fontSize: 16
    },
    smallContentText: {
        fontSize: 14,
    },
    mediumContentText: {
        fontSize: 16
    },
    contentWrapper: {
        flexDirection: 'row'
    },
    icon: {
        justifyContent: 'center'
    },
    separator: {
        width: 2,
        backgroundColor: '#aaa',
        margin: 5
    }
});
