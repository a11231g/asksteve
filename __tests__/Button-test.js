import React from 'react';
import Button from '../src/components/Button/Button';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
});
