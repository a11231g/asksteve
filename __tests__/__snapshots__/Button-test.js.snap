// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#1e87f0",
      "borderRadius": 3,
      "height": 40,
      "justifyContent": "center",
      "opacity": 1,
      "padding": 8,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "textAlign": "center",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "textAlign": "center",
          },
          undefined,
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#eee",
              "fontSize": 16,
            },
            Object {
              "fontSize": 14,
            },
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;
